core/txpool/txpool.go

```diff
@@ -688,7 +688,21 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 	if l1Cost := pool.l1CostFn(tx.RollupDataGas(), tx.IsDepositTx()); l1Cost != nil { // add rollup cost
 		cost = cost.Add(cost, l1Cost)
 	}
-	balance := pool.currentState.GetBalance(from)
+
+	metaTxParams, err := types.DecodeAndVerifyMetaTxParams(tx)
+	if err != nil {
+		return err
+	}
+	var balance *big.Int
+	if metaTxParams != nil {
+		if metaTxParams.ExpireHeight < pool.chain.CurrentBlock().Number.Uint64() {
+			return types.ErrExpiredMetaTx
+		}
+		balance = pool.currentState.GetBalance(metaTxParams.GasFeeSponsor)
```

Comment:
What happens when there is not enough balance in st.msg. MetaTxParams. GasFeeSponsor

Reply:
In line 706, `balance` will be checked against tx cost.